### PR TITLE
DEV-4703: Add BE validation errors to ResetPassword

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a
#### What's this PR do?
- Add BE password validation error to ResetPassword
#### Why are we doing this? How does it help us?
- If ResetPassword fails, show the user the error so that they can fix it
#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
- yes
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
- no
#### Have automated unit tests been added? If not, why?
- yes
#### How should this change be communicated to end users?
- n/a
#### Are there any smells or added technical debt to note?
- no
#### Has this been documented? If so, where?
- no
#### What are the relevant tickets? Add a link to any relevant ones.
- https://news-revenue-hub.atlassian.net/browse/DEV-4703
#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:
- no
#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
- no

### CODE REVIEW

File: spa/src/components/account/ResetPassword/ResetPasswordForm.tsx [hunk 1]
```diff
@@ -6,13 +6,15 @@ import { Root } from './ResetPasswordForm.styled';
...
 const ResetPasswordFormPropTypes = {
   disabled: PropTypes.bool,
-  onSubmit: PropTypes.func.isRequired
+  onSubmit: PropTypes.func.isRequired,
+  passwordError: PropTypes.string
 };
 
 export interface ResetPasswordFormProps extends InferProps<typeof ResetPasswordFormPropTypes> {
   onSubmit: (password: string) => void;
+  passwordError?: string;
```
Comment (nrh-cklimas) [inline spa/src/components/account/ResetPassword/ResetPasswordForm.tsx:15]: I think InferProps gives us this--no need to specify, unlike functions where we need to spec the arguments.

File: spa/src/components/account/ResetPassword/ResetPassword.test.tsx
Comment (nrh-cklimas) [inline spa/src/components/account/ResetPassword/ResetPassword.test.tsx:1]: We should add tests here exercising the network aspects of ResetPassword.

- Sends to correct endpoint using POST
- Handles success correctly
- Passes errors down to ResetPasswordForm 👈 this is the part we really want to test, but coverage on the component isn't good in main

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]

File: spa/src/components/account/ResetPassword/ResetPasswordForm.tsx (hunk 1 above)
Comment (GuiMend) [reply spa/src/components/account/ResetPassword/ResetPasswordForm.tsx:15]: The `InferProps` does `passwordError = null | undefined | string` and I was overriding to not accept null

Comment (nrh-cklimas) [reply spa/src/components/account/ResetPassword/ResetPasswordForm.tsx:15]: Hm--I took this line out locally and TypeScript doesn't complain. It's not a big deal either way.

File: spa/src/components/account/ResetPassword/ResetPassword.test.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    await screen.getByRole('button', { name: 'Reset Password' }).click();
+
+    await waitFor(() => expect(axiosMock.history.post.length).toBe(1));
+    expect(axiosMock.history.post[0].data).toBe('{"token":null,"password":"mock-password"}');
+    expect(screen.getByText(RESET_PASSWORD_SUCCESS_TEXT)).toBeInTheDocument();
+  });
+
+  describe('when submitting returns an error', () => {
+    it('should pass validation errors to the form', async () => {
```
Comment (nrh-cklimas) [inline spa/src/components/account/ResetPassword/ResetPassword.test.tsx:34]: One last coverage thing--should test what happens if the endpoint returns a network error. That part appears uncovered in `npm run test:coverage`.

Comment (GuiMend) [reply spa/src/components/account/ResetPassword/ResetPassword.test.tsx:46]: @nrh-cklimas , I'm not sure how to cover line 48-49. I actually don't think that is ever used. What do you say we remove it?

**Review by nrh-cklimas — APPROVED** [review]

File: spa/src/components/account/ResetPassword/ResetPassword.test.tsx (hunk 2 above)
Comment (nrh-cklimas) [reply spa/src/components/account/ResetPassword/ResetPassword.test.tsx:46]: Noting for future reference that we talked this through on Slack.

**Pull request merged**